.kiota/apimanifest.json

```diff
@@ -0,0 +1,79 @@
+{
+  "applicationName": "application",
+  "apiDependencies": {
+    "Calendar": {
+      "apiDescriptionUrl": "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/graphexplorer.yaml",
+      "apiDeploymentBaseUrl": "https://graph.microsoft.com/v1.0/",
+      "requests": [
+        {
+          "method": "GET",
+          "uriTemplate": "me/calendar/events"
+        }
+      ],
+      "extensions": {
+        "x-ms-kiota-hash": "29814D37A98A70300C212A353CCD01919F5B58D6C8F741C4948E4B7A5FDAA44B",
+        "x-ms-kiota-version": "1.20.0-preview.202410170001"
+      }
+    },
+    "Contacts": {
+      "apiDescriptionUrl": "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/graphexplorer.yaml",
+      "apiDeploymentBaseUrl": "https://graph.microsoft.com/v1.0/",
+      "requests": [
+        {
+          "method": "GET",
+          "uriTemplate": "me/contacts"
+        }
+      ],
+      "extensions": {
+        "x-ms-kiota-hash": "21808D91F876D396FF383233E2CC18B4DD82CDFE2A403B155CBAEC18808808E2",
+        "x-ms-kiota-version": "1.20.0-preview.202410170001"
+      }
+    },
+    "DriveItem": {
+      "apiDescriptionUrl": "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/graphexplorer.yaml",
+      "apiDeploymentBaseUrl": "https://graph.microsoft.com/v1.0/",
+      "requests": [
+        {
+          "method": "GET",
+          "uriTemplate": "drive/root/children/{driveItem-id}/content"
+        }
+      ],
+      "extensions": {
+        "x-ms-kiota-hash": "883D3BE04064138511FB79FDB1C2A7883D9E8F492481A04F0D55AA75B63CA642",
+        "x-ms-kiota-version": "1.20.0-preview.202410170001"
+      }
+    },
+    "Messages": {
+      "apiDescriptionUrl": "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/graphexplorer.yaml",
+      "apiDeploymentBaseUrl": "https://graph.microsoft.com/v1.0/",
+      "requests": [
+        {
+          "method": "GET",
+          "uriTemplate": "me/messages"
+        },
+        {
+          "method": "POST",
+          "uriTemplate": "me/messages"
+        }
+      ],
+      "extensions": {
+        "x-ms-kiota-hash": "C8360ADB984F44449D607A0517A4AE13A7BF0BF2F53FA7AA147257F49D6BBF8F",
+        "x-ms-kiota-version": "1.20.0-preview.202410170001"
+      }
+    },
+    "Astronomy": {
+      "apiDescriptionUrl": "https://raw.githubusercontent.com/zengin/openapi-directory/zengin/nasa/APIs/nasa.gov/apod/1.0.0/openapi.yaml",
```

Comment:
I think we should not take a dependency on my account here. Is there any chance this file can be moved to a Microsoft owned location? I think I needed a copy for adding auth details.

Reply:
Sure, we can point it somewhere else. Do you remember where you took it from?

Reply:
The origin is `APIs-guru/openapi-directory` and this is the change I had to make to work with operationId based function name generation in Semantic Kernel: https://github.com/APIs-guru/openapi-directory/compare/main...zengin:openapi-directory:zengin/nasa

Reply:
Thanks for pointing this out.
APIs.guru dot NOT accept PR on description, this is because they sync it from the origin.
In this case the origin is gone, NASA decided to create a giganormous document with all the APIs they have instead as far as I can tell https://github.com/nasa/api-docs/blob/master/assets/json/apis.json. This makes it an invalid OAS description unless we do additional processing to extract the "swagger" node...
The interest or an upstream PR is limited, we wouldn't be able to consume the document as is, and APIs guru still won't be able to sync the update.
I think our best bet is to create a local copy, which I've placed in `dotnet\samples\Concepts\Resources\Plugins\OpenAPI\NASA\apod.yaml` @SergeyMenshykh please let me know if this path works for you as well.